packages/gatsby/src/services/calculate-dirty-queries.ts

```diff
@@ -5,11 +5,54 @@ import { assertStore } from "../utils/assert-store"
 
 export async function calculateDirtyQueries({
   store,
+  websocketManager,
+  currentlyHandledPendingQueryRuns,
 }: Partial<IQueryRunningContext>): Promise<{
   queryIds: IGroupedQueryIds
 }> {
   assertStore(store)
   const state = store.getState()
   const queryIds = calcDirtyQueryIds(state)
-  return { queryIds: groupQueryIds(queryIds) }
+
+  let queriesToRun: Array<string> = queryIds
+
+  if (
+    // eslint-disable-next-line
+    false &&
```

Comment:
actually - for this PR I wanted to keep it that way, merge into yours and merge into master. And separate PR against master together with runtime changes uncomment that ;) but I can remove it from here

Reply:
I'd add a TODO or FIXME comment then, to make it clear it's not a mistake

Reply:
Oh, just saw the comment underneath!

Reply:
I will be doing minor updates here soon - mostly just addressing that `NIT` about single action not having to be in array - do you think I should remove this code piece from here?

Primary purpose of it was just to show how the `pendingQueryRuns` would be actually used in `calculateDirtyQueries`, but otherwise it's dead code for now until rest of things ship

Reply:
I dropped this dead code in https://github.com/gatsbyjs/gatsby/pull/28111/commits/c82842b9611ed1a8b937d2a41754aa779349ebc6 - no point in having it in - it was just for illustration how the `pendingQueryRuns` will be actually used in follow up PRs